### PR TITLE
Define Platform and OutputPaths

### PR DESCRIPTION
**Pull request by gkhanna79**
Define Platform and Output Path variables ahead of including any traversal target files.

### CODE REVIEW

**Comment by gkhanna79** [comment]
@weshaggard @ellismg PTAL  - this is to address the VSO OSX issue.

File: dir.props [hunk 1]
```diff
@@ -69,8 +69,19 @@
...
+    <Platform Condition="'$(Platform)' == ''">$(BuildArch)</Platform>
+    <Platform Condition="'$(Platform)' == 'amd64'">x64</Platform>
+    
   </PropertyGroup>
 
+  <!-- Output paths -->
+  <PropertyGroup>
+    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)' == ''">$(RootBinDir)obj</BaseIntermediateOutputPath>
+    <IntermediateOutputPath Condition="'$(IntermediateOutputPath)' == ''">$(BaseIntermediateOutputPath)\$(BuildOS).$(BuildArch).$(Configuration)</IntermediateOutputPath>
+    <OutputPath Condition="'$(OutputPath)' == ''">$(BaseIntermediateOutputPath)\$(BuildOS).$(BuildArch).$(Configuration)</OutputPath>
```
Comment (weshaggard) [inline dir.props]: Why is the OutputPath under Intermediate? Generally they are parallel, see https://github.com/dotnet/corefx/blob/master/dir.props#L381

Comment (gkhanna79) [inline dir.props]: @weshaggard I have picked this up from mscorlib.csproj. If it is not correct there, let me know and I will fix both the locations.

Comment (weshaggard) [inline dir.props]: @AlexGhiondea any ideas why they are set that way for mscorlib? I'm not sure what FinalOutputPath is used for but OutputPath is generally the finial bin-place location which should usually be parallel to the intermediate directory.

Comment (gkhanna79) [inline dir.props]: I think this is so because of binplacing that happens in mscorlib build process from OutputPath to FinalOutputPath

Comment (AlexGhiondea) [inline dir.props]: If I remember correctly, what @gkhanna79 is saying is right :)

**Comment by gkhanna79** [comment]
@weshaggard Are you good with this change?

**Comment by weshaggard** [comment]
I'm good my comment was more of a question about why these are that way.

**Pull request merged**